torch/nn/modules/module.py

```diff
@@ -616,26 +616,21 @@ def register_backward_hook(
     ) -> RemovableHandle:
         r"""Registers a backward hook on the module.
 
-        .. warning ::
-
-            The current implementation will not have the presented behavior
-            for complex :class:`Module` that perform many operations.
-            In some failure cases, :attr:`grad_input` and :attr:`grad_output` will only
-            contain the gradients for a subset of the inputs and outputs.
-            For such :class:`Module`, you should use :func:`torch.Tensor.register_hook`
-            directly on a specific input or output to get the required gradients.
-
         The hook will be called every time the gradients with respect to module
         inputs are computed. The hook should have the following signature::
 
-            hook(module, grad_input, grad_output) -> Tensor or None
+            hook(module, grad_input, grad_output) -> tuple(Tensor) or None
 
-        The :attr:`grad_input` and :attr:`grad_output` may be tuples if the
-        module has multiple inputs or outputs. The hook should not modify its
-        arguments, but it can optionally return a new gradient with respect to
-        input that will be used in place of :attr:`grad_input` in subsequent
-        computations. :attr:`grad_input` will only correspond to the inputs given
-        as positional arguments.
+        The :attr:`grad_input` and :attr:`grad_output` are tuples. The hook should
+        not modify its arguments, but it can optionally return a new gradient with
+        respect to the input that will be used in place of :attr:`grad_input` in
+        subsequent computations. :attr:`grad_input` will only correspond to the inputs given
+        as positional arguments. Entries in :attr:`grad_input` and :attr:`grad_output` will
```

Comment:
May want to extend this sentence with an extra clause clarifying that kwarg-only arguments will not appear in the hook.

Reply:
This is true for all the nn.Module hooks actually. I will update.